### DISCUSSION TITLE
Support for inline code for Node JS 14?

### DISCUSSION THREAD

**Discussion by segura27** [aws/aws-cdk#22150] [Q&A]
I'm trying to move away from Node JS 12 but one of my Lambdas is throwing: `Error: Inline source not allowed for nodejs14.x` when I try to upgrade it to run on Node 14.

Is there a configuration change I can make to avoid this error or am I locked out of Node 14 for Lambdas with inline code?

I can't seem to find any documentation on this.

**Comment by pahud** [comment]
node14 should be supported 
https://github.com/aws/aws-cdk/blob/74318c7d22bfc00de9e005f68a0a6aaa58c7db39/packages/%40aws-cdk/aws-lambda/lib/runtime.ts#L81

Try deploy a lambda function like this. It should work.

```ts
    new lambda.Function(this, 'Func', {
      runtime: lambda.Runtime.NODEJS_14_X,
      code: lambda.Code.fromInline('foo'),
      handler: 'index.handler',
    });
```
